### DISCUSSION TITLE
apbs on large repeating biological assemblies

### DISCUSSION THREAD

**Discussion by Zlotka** [Electrostatics/apbs#237] [Q&A]
Hi,
I have been trying to do the apbs calculation of the whole icosahedral virus capsid (60 x asymmetric subunit), but it always gives an error that I have concatenated files (which I do as it is 60x the same protein, different coordinates) and does not run the calculations. 
If I use PDB2PQR and ticking the box to add whitespace it does not open in ChimeraX (which I use to obtain coordinates for the full capsid). I tried adding the whitespace in Notepad++, but it is not optimal as the file is large and it crashes every time I use find/replace.
Could I work around this issue somehow? 

I am using the web browser version.

Thank you

**Comment by jbardhan** [comment]
Hi,

I appreciate that the file size for the full capsid makes things even harder! 

Could you make a partial capsid, say with just two asymmetric subunits, and upload the files here?  That will be extremely helpful.

Best regards,
Jay

**Reply by Zlotka** [reply]
[6gzv-sym.zip](https://github.com/Electrostatics/apbs/files/8830313/6gzv-sym.zip)

I put 2 asymmetric subunits in .pdb and .pqr (no whitespace).

KR
Zlatka

**Reply by jbardhan** [reply]
Thank you!  I will have a look at this but I might not get to it until the weekend.

Best regards,
Jay

**Reply by jbardhan** [reply]
Thanks for your patience.

I've had a look and here's my quick take.  The PDB shows the different subunits as different models rather than as being "one molecule," and the PQR looks the same.  This may be confusing APBS.  

For a very ugly hack, you might try to delete the PQR lines in between the two sets of ATOM records, and see what happens:
"ENDMDL
CONECT11821118261182211816
CONECT1182211821118271182511823
CONECT118231182211824
CONECT1182411823
CONECT1182511822118281182911830
CONECT1182611821
CONECT1182711822
CONECT1182811825
CONECT1182911825
CONECT1183011825
MODEL        2
"

I'll try a simpler, smaller PQR with multiple models to see if that reproduces the behavior; not sure if I will get to it today.

Does this provide some insights?  Can you share any additional details about the methods/code you're using to generate these?

Jay

**Reply by Zlotka** [reply]
Thank you for the suggestion. I will try that.

I usually obtain .pdb files using E-Tasser and adjust them in WinCoot (to cryoEM density map). I used PDB2PQR to generate a .pqr file. In ChimeraX I use the command to symmetrize the .pqr file to icosahedral symmetry (sym #0 i,222 copies true). This generates 60 copies of the asymmetric subunit as models with different coordinates. I save all the models in the same .pqr file to be used with apbs. I have not figured out yet how to generate the whole capsid as one molecule in ChimeraX or Chimera so deleting the lines manually might be the easiest.

Zlatka

**Reply by Zlotka** [reply]
![ch-photo](https://user-images.githubusercontent.com/55322634/172358092-d594f43e-5350-4888-bcff-a3b94bfb5540.JPG)
P.S. this is how it looks like after deleting those lines. The job ran, but the results don't make sense.

**Reply by Zlotka** [reply]
I get the same result when I use the combined models in .pqr file (command: combine #0 #1 in ChimeraX). Is it because I am using input files for apbs (.in, .pqr, .log) which were generated from 1 asymmetric subunit?
I get this output in apbs.stderr.txt:

Unusually large potential values
    detected on the focusing boundary!
    Convergence not guaranteed for NPBE/NRPBE calculations!

**Reply by jbardhan** [reply]
Thank you! It seems like we're making progress.

I'll try to run at least one test today myself.  Here are my immediate thoughts:

1) Are you saying that you are taking an APBS.in file that you used for 1 asymmetric subunit, and merely replacing the PQR file with the 2-subunit PQR?  If so, that would definitely be likely to cause the issue you are seeing with unusually large potential values.  If you can share the .in and .pqr files giving you this, I'll take a look.

2) Am I correct to interpret your workflow as the following:
E-tasser -> raw 1-subunit.pdb file  --> WinCoot --> refined structure --> PDB2PQR --> 1-subunit.pqr file --> XChimera --> 60-subunit.pqr ?

Thanks very much,
Jay

**Reply by Zlotka** [reply]
Hi,
1. You are correct. I am using .in file made from 1 subunit. If I put 2 or more subunits to PDB2PQR it will clean up the file to contain only one. I don't understand the .in file well enough to edit anything.
[apbs_input_files.zip](https://github.com/Electrostatics/apbs/files/8876431/apbs_input_files.zip)
2. That's correct!

Thank you
